core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -134,7 +134,7 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
       // Periodically revive offers to allow delay scheduling to work
       val reviveIntervalMs = conf.get(SCHEDULER_REVIVE_INTERVAL).getOrElse(1000L)
 
-      reviveThread.scheduleAtFixedRate(() => Utils.tryLogNonFatalError {
+      reviveThread.scheduleWithFixedDelay(() => Utils.tryLogNonFatalError {
         Option(self).foreach(_.send(ReviveOffers))
```

Comment:
I think it's meaningless to replace for this kind of task whose intentional action happens asynchronically. For example, in this case, sending the message `ReviveOffers` should finish quickly but the intentional action `reviveOffers` may take a longer time. I guess the *delay* here you want to deal with is from the former one rather than the latter.

Reply:
> I think it's meaningless to replace for this kind of task whose intentional action happens asynchronically. For example, in this case, sending the message `ReviveOffers` should finish quickly but the intentional action `reviveOffers` may take a longer time. I guess the _delay_ here you want to deal with is from the former one rather than the latter.

If there is a delay in Sender side, avoid send `ReviveOffers` twice, then receive side  don't need to call `makeOffers` twice in  repeatedly.

Reply:
> I think it's meaningless to replace for this kind of task whose intentional action happens asynchronically. For example, in this case, sending the message `ReviveOffers` should finish quickly but the intentional action `reviveOffers` may take a longer time. I guess the _delay_ here you want to deal with is from the former one rather than the latter.

How about retained change?